packages/unlock-js/src/PublicLock/v4/purchaseKey.js

```diff
@@ -13,11 +13,13 @@ import { approveTransfer, getErc20Decimals, getAllowance } from '../../erc20'
  * @param {function} callback invoked with the transaction hash
  */
 export default async function (
-  { lockAddress, owner, keyPrice, erc20Address, decimals },
+  { lockAddress, owner, keyPrice, erc20Address, decimals, swap },
   transactionOptions = {},
   callback
 ) {
   const lockContract = await this.getLockContract(lockAddress)
+  const unlockSwapPurchaserContract =
```

Comment:
TBH I am not sure we need to support swap and purchase for locks that are this old. If that creates too much trouble let's just skip adding support for this version

Reply:
Already added code so I will leave it here unless there are issues and it doesn't work in testing.